spec/classes/profile/solr_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018-2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'nebula::profile::solr' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile }
+
+      context 'with package' do
```

Comment:
This all mostly looks good. My only comment is that this isn't how contexts are meant to be used. A context implies something external to the profile, i.e. a parameter change or a fact about the machine. For example, [kubernetes_spec.rb][1] has a number of contexts, and you can see how each sets a parameter, creating the described context.

[1]: https://github.com/mlibrary/nebula/blob/master/spec/classes/profile/kubernetes_spec.rb

A context is something about the outside world, like "the machine has less than 2G of ram" or "the hostname is set to default.invalid." It isn't meant to be used for categorical grouping of conceptually related tests. The way we do that is just to put them next to each other.

So I'd like to see the contexts removed. To be honest, I'd also get rid of the array loops, since I don't feel like there's much of a need to DRY out

```ruby
it { is_expected.to contain_package('openjdk-8-jre-headless') }
it { is_expected.to contain_package('solr') }
it { is_expected.to contain_package('lsof') }
```

I personally prefer to spell out each test, especially when the tests are so straightforward. When each test is a few lines long, this gets onerous, but with one-line tests, I feel the repeated assertions add clarity that makes up for the minor repetition. But that's a matter of taste, so I'll leave that decision to you.

Reply:
I replaced context. And thanks for clearing it up. I took another look at context and it is making more sense now. :)